js/views/userPage/UserPage.js

```diff
@@ -65,25 +78,34 @@ export default class extends BaseVw {
   }
 
   selectTab(targ) {
-    let tabViewName = targ.data('tab');
-    let tabView = this.tabViewCache[tabViewName];
-    const tabViewType = tabViewName; // the original view name is passed in for the Follow view
-
-    this.$tabTitle.text(tabViewName);
+    // if an invalid targ is passed in, set it to Store
+    if (!this.tabViews[targ] && targ !== 'Following' && targ !== 'Followers') {
+      targ = 'Store'; // eslint-disable-line no-param-reassign
+    }
 
-    if (tabViewName === 'Followers' || tabViewName === 'Following') tabViewName = 'Follow';
+    let tabView = this.tabViewCache[targ];
+    const tabOptions = { ownPage: this.ownPage, model: this.model };
 
     if (!this.currentTabView || this.currentTabView !== tabView) {
+      this.$tabTitle.text(targ);
+      // add tab to history
+      app.router.navigate(`${this.model.id}/${targ}`, {
+        trigger: false,
+        replace: false,
```

Comment:
I believe trigger and replace are false by default, so no need to pass them in.

Although, this brings up a UX question (maybe for Mike), do we want each tab to be a separate item in history?